### DISCUSSION TITLE
Analysis on MacStudio then move back to Raspberry Pi

### DISCUSSION THREAD

**Discussion by hanshafner** [immich-app/immich#8949] [Q&A] [answered]
Hi there,

is it possible to make the library portable in the sense that it's one directory containing everything to run the docker container and then move that from one system to another?

Especially I'd like to use my MacStudio to do all the face detection / object detection machine learning stuff but then move and host the library on the raspberry pi again.

**Comment by bo0tzz** [comment]
https://immich.app/docs/guides/remote-machine-learning may be easier to accomplish what you want.

**Comment by hanshafner** [comment]
Thank you! I did not know about this function, sounds exactly like what I want.

I will try to set that up.

**Reply by hanshafner** [reply]
Ah, that was rather simple.

But the original question still stands, because the Pi is just also very slow in transcoding and generating thumbs and all that.

So - is it possible to have the .yml, all my pictures and all data from immich in one directory to easily move it between systems, or am I being way too naive and the architectures will not be as easy to change?

**Reply by mmomjian** [reply, marked as answer]
You can follow the instructions in the docs to backup and restore to a new system: https://immich.app/docs/administration/backup-and-restore/

**Reply by hanshafner** [reply]
Oh my ... now I feel really stupid. Of course!

The added benefit being that I can practice backup and restore... something I do way too little, especially the restore part...

Thank you!

I've done it once this morning, moving over to the MacStudio worked great (analysis is soooo quick), moving back I made a couple of stupid mistakes (mainly trying to save transfer time and space with the original images...) - but that's ok, it is all experimental at this point and I get more confident with each mistake...
